### PR TITLE
Build kernels during the image-building process

### PR DESCRIPTION
**Pull request by djmitche**
Mostly as described in taskcluster/taskcluster#3574, this makes the monopacker stuff a bit more regular and figures out when it should and should not rebuild the kernel.

In its draft form it doesn't actually rebuild the kernel, but this works for GCP!

This is based on #87 so some of these commits will vanish when that lands.

This won't quite fix the issue, as I'll need to update a few other places with the new builder names.

### CODE REVIEW

**Comment by imbstack** [comment]
Ok, [this](https://linux.die.net/lkmpg/x380.html) explains what we need to do to get the modules compiled for the proper kernel.

tl;dr `EXTRAVERSION` and `SUBLEVEL` in the makefile needs to be updated to match whatever the original person who compiled had in there.

I've done this on my machine manually and it seems to work fine. Tomorrow I'll update this PR to do it automatically

**Comment by djmitche** [comment]
Nice!  I bet the Debian Makefile stuff does that somewhere and it just didn't get run in this process.

File: scripts/ubuntu-bionic/00-setup.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        esac
+        case $NUM_LOOPBACK_VIDEO_DEVICES in
+            0) BUILD_V4L2LOOPBACK=false ;;
+            *) BUILD_V4L2LOOPBACK=true ;;
+        esac
+        ;;
+    aws)
+        case $NUM_LOOPBACK_AUDIO_DEVICES in
+            0) SETUP_SND_ALOOP=false ;;
+            32) SETUP_SND_ALOOP=true ;;
```
Comment (djmitche) [inline scripts/ubuntu-bionic/00-setup.sh:37]: I think this needs OLD_KERNEL=true as well?

File: scripts/ubuntu-bionic/00-setup.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            *) SETUP_SND_ALOOP=true OLD_KERNEL=true ;;
+        esac
+        case $NUM_LOOPBACK_VIDEO_DEVICES in
+            0) BUILD_V4L2LOOPBACK=false ;;
+            *) BUILD_V4L2LOOPBACK=true OLD_KERNEL=true ;;
```
Comment (djmitche) [inline scripts/ubuntu-bionic/00-setup.sh:42]: For 32 devices, can we get away without an OLD_KERNEL?

**Comment by djmitche** [comment]
https://github.com/taskcluster/taskcluster/issues/3574 has a bunch of scenarios that we should test/support.

 * *community / docker-worker / gcp* - needs no special treatment -- just enable snd-aloop in `/etc/modules`
 * *community / docker-worker / aws* - unused (but let's treat it the same as *firefox-ci / docker-worker / aws*)
 * *firefox-ci / docker-worker / gcp* - needs v4l2loopback module built (taskcluster/monopacker#83), snd-aloop just needs to be enabled
 * *firefox-ci / docker-worker / aws* - needs v4l2loopback and snd-aloop modules built
 * *firefox-ci / docker-worker / aws-metal* - needs v4l2loopback and snd-aloop modules built, both patched to support 96 devices
 
You've learned that we don't need special metal support.  And, firefox-ci is still running substantially on the old (non-monopacker, 14.04) docker-worker, which doesn't even appear on this chart.  So we can probably limit ourselves to the desired end-state for firefox-ci, which is *firefox-ci / docker-worker / gcp*.  That means really all we have to care about is

 * *community / docker-worker / gcp* - needs no special treatment -- just enable snd-aloop in `/etc/modules`
 * *firefox-ci / docker-worker / gcp* - needs v4l2loopback module built (taskcluster/monopacker#83), snd-aloop just needs to be enabled

It's easy enough to keep and test support for generic aws and azure support, but let's do so without anything that's not otherwise supported, so something like:

 * *community / docker-worker / aws* - unused, no devices support
 * *community / docker-worker / azure* - unused, no devices support

and add a note in the case/esac that the -aws kernel does not support snd-aloop.  There are probably some other pearls of wisdom that can be left in comments.

**Comment by imbstack** [comment]
Ok, based on that analysis I've just removed snd-aloop support from aws. I believe this is ready. I can't request review from you since it is technically your PR :1st_place_medal:

File: scripts/ubuntu-bionic/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+* `CLOUD` - the cloud being built for (`azure`, `aws`, or `google`)
+* `NUM_LOOPBACK_AUDIO_DEVICES` - number of snd-aloop devices (for docker-worker's loopbackAudio support) to support (0 to disable)
+* `NUM_LOOPBACK_VIDEO_DEVICES` - number of v4l2loopback devices (for docker-worker's loopbackVideo support) to support (0 to disable)
+* `V4L2LOOPBACK_VERSION` - version of the v4l2loopback module to install (default in the script)
+
+Note that the kernel version used is that of the base image.
+
+## Behaviors
+
+* Sets up the kernel for the desired loopback audio and video devices.  Some combinations require recompiling the kernel, and in some cases (GCP) this is not currently supported.
```
Comment (djmitche) [inline scripts/ubuntu-bionic/README.md:18]: Maybe just "not all combinations are supported"?

File: scripts/ubuntu-bionic/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+* Installs some basic packages required to build things (build-essentials, gnupg, curl)
+* Installs some "misc packages"; see `30-packages.sh` for the list.  Generally these are small and fine to install unconditionally.
+* Installs `python-statsd` for reasons that are lost to history
+* Installs a fix for docker networking (https://github.com/moby/libnetwork/issues/1090)
+
+## Notes
+
+The kernel variants are quite different, even at the level of how they are packaged.
+As we currently only need to recompile on AWS, recompilation is not supported for the GCP variant.
+If this becomes necessary, some work will be required to figure out how to rebuild the GCP variant.
```
Comment (djmitche) [inline scripts/ubuntu-bionic/README.md:29]: Maybe include some of the notes from this PR here?

**Review by djmitche — COMMENTED** [review]
Looks good.  The README is probably a good place to stash some wisdom, but good to merge with that changed.

**Comment by imbstack** [comment]
I'll start testing these images later today

**Pull request merged**